src/ansys/fluent/core/session.py

```diff
@@ -343,8 +346,87 @@ def download(self, file_name: str, local_file_name: Optional[str] = None):
         FileNotFoundError
             If the remote file does not exist.
         """
-        if self.file_service:
-            if self.file_service.file_exist(file_name):
-                self.file_service.download_file(file_name, local_file_name)
-            else:
-                raise FileNotFoundError("Remote file does not exist.")
+        self._upload_download_helper(
+            is_upload=False, file_name=file_name, local_file_name=local_file_name
+        )
+
+    def _wait_for_file(self, file_name: str):
+        """Wait for file to get ready for upload or download.
+
+        Parameters
+        ----------
+        file_name : str
+            File name
+        Raises
+        ------
+        FileNotFoundError
+            If a file does not exist.
+        """
+        start_time = time.time()
+        max_wait_time = sys.maxsize
+        while (time.time() - start_time) < max_wait_time:
+            if self.file_service.file_exist(os.path.basename(file_name)):
+                break
+            max_wait_time -= 1
+            time.sleep(3)
+        else:
+            raise FileNotFoundError(f"{file_name} does not exist.")
```

Comment:
`max_wait_time = sys.maxsize` is basically the same as an infinite amount of time, is that correct? 

In that case, the `raise FileNotFoundError(f"{file_name} does not exist.")` below is realistically never going to be raised.

Assuming that waiting for up to one minute should be more than enough, my suggestion would be something like:
```suggestion
        from ansys.fluent.core.utils.execution import timeout_loop
        file_exists = timeout_loop(
            self.file_service.file_exist, 
            timeout=60, 
            args=(os.path.basename(file_name),),
            idle_period=3
        )
        if not file_exists:
            raise FileNotFoundError(f"{file_name} does not exist.")
```
Which will check if the file exists every three seconds until `file_exist` returns true (or fail if timeout of 60 seconds is reached).

Reply:
It is the same case whether we use current implementation or `timeout_loop`. We don't know what is the ideal time limit to upload or download file due to following uncertainties. Sean has already mentioned this. 

1. Internet Speed
2. Size of file

We will also use `sys.maxsize` in `timeout_loop` due to above mentioned uncertainties. (60 seconds is not suitable, 100 was prior to this, but we changed it to `sys.maxsize`)

Reply:
Do we plan on changing the max wait time from `sys.maxsize` to something that can realistically ever be reached? Otherwise there is no point to including `raise FileNotFoundError(f"{file_name} does not exist.")` here, as it is effectively unreachable.

And yes, `timeout_loop` is just a convenience function that we already have that achieves what you are adding here. If you want to use this code block rather than that function, that is fair, but why `time.sleep(3)` and then `max_wait_time -= 1`? Wouldn't it make more sense to reduce the wait time by the same amount of time that you set it to sleep for?

Reply:
@raph-luc we decided not to bother with a finite upper limit since we cannot quantify it.

Reply:
but you're correct about the unreachable exception

Reply:
@raph-luc yes we can update this code according to above suggestion. So it will avoid dependency on time limit. Thank you.

Reply:
@raph-luc  `but why time.sleep(3) and then max_wait_time -= 1?`

It's ok to have `time.sleep(1)` and `max_wait_time -= 1` for small amount of time like 60 seconds, but in current case it will be much more than 60 seconds. In Ansys Lab we don't want to query status of file for every second therefore we have made it delay of 3 seconds. There is no standard relationship between `time.sleep()` and decrement operator, we are using it according to our purpose.